### PR TITLE
Fix build on OS X

### PR DESCRIPTION
**Pull request by jjrv**
`auto.gypi` and `auto-top.gypi` are specific to the user's environment so it would be better to remove them from Github and call `autogypi` from an `install` script, which could also compile libui.

The `libui` compiled library file name is different on OS X. It seems with these changes it works, also on Linux.

### CODE REVIEW

**Comment by jjrv** [comment]
Seems to be more work than I thought to get Travis configured for libui. I'll rebase this thing until it works, ignore the PR until then...

**Comment by jjrv** [comment]
OK I'm done. So the first 3 commits would be nice to get merged for using libui-node on OS X and easier installation in general. The last 2 were an attempt to get Travis to like my PR. It's not going to happen, but now it gets far enough to run `ava` and it looks like the failures are due to actual unimplemented features in the library, instead of installation problems.

The container-based environment on Travis is so old I just couldn't get libui to work, so now it uses the beta Trusty environment. Otherwise I can't get Travis to do anything reasonable. I hope soon they'll make Trusty work with containers as well.

File: example.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const libui = require('.');
+const libui = require('./index.js');
```
Comment (parro-it) [inline example.js]: Why this change? `require('.')` should be independent by the cwd and equivalent to `require('./index.js')`

File: test.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import test from 'ava';
-import libui from '.';
+import libui from './index';
```
Comment (parro-it) [inline test.js]: Why this change? require('.') should be independent by the cwd and equivalent to require('./index.js')

File: example.js (hunk 1 above)
Comment (jjrv) [inline example.js]: I was testing with Node.js 0.10 and that was the only thing that caused it to fail, so I changed it and forgot to add a note... It seems the automatic index.js fetching is a later feature. Hopefully nobody actually uses such an old version though.

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     }
   ],
   "scripts": {
-    "test": "ava && xo",
+    "test": "LD_LIBRARY_PATH=./libui/build/out ava && xo",
```
Comment (parro-it) [inline package.json]: I was not able to find a way to remove the LD_LIBRARY_PATH. Seems silly to need this

Comment (jjrv) [inline package.json]: It's not needed on OS X. The earlier libui.js has this in its `binding.gyp`:

`"libraries": ["-Wl,-rpath, ../libui/out/libui.so.0"]`

Apparently that might make it work without `LD_LIBRARY_PATH` on some Linux distributions, but not on my Ubuntu. The equivalent for OS X in this PR does work. I spent some time trying to understand a linker option `-Wl,--enable-new-dtags` that might have fixed the issue on Linux but ultimately it didn't help.

File: index.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
 const nbind = require('nbind');
-const binding = nbind.init();
+const binding = nbind.init(__dirname);
```
Comment (jjrv) [inline index.js]: Probably this nbind feature is not emphasized enough in the docs... Without the `__dirname` parameter it looks for compiled binaries under `process.cwd()` which will fail if run from outside the root of that specific package, particularly if others are using your bindings in their project.

**Comment by parro-it** [comment]
Thank you so much @jjrv, great work! 
I'll merge the PR and do some changes on tests to make them pass.
The tests was passing in a an old release, before I start using nbind.
If you want, we can cooperate to implements widgets events handling in `libui-node` using `nbind`.
I can help you to solve the [Callbacks lifetime](https://github.com/charto/nbind/issues/15) issue on `nbind`.

**Comment by jjrv** [comment]
Thanks! Your project is the perfect platform to verify the callback support.

**Comment by parro-it** [comment]
Tests are passing succesfully on Travis now. I have to add another package to Travis conf to add an headless graphic env.
